lib/csu/common-cheri/crt_init_globals.c

```diff
@@ -32,53 +32,22 @@
 #include <stdint.h>
 #include <stdlib.h>
 
+#ifdef PIC
+#error "PIEs never need to initialise their own globals"
```

Comment:
Note that csu is weird, it uses -fPIC + -DPIC for Scrt1, but Scrt1 is for PIEs, DSOs don't have a crt1 at all as that's where _start lives. Should really be -fPIE + -DPIE. But that's why the message says "PIEs" here not "Shared libraries" as one might expect from the ifdef.

Reply:
It would not surprise me if this was due to ignorance, and perhaps at the time the csu Makefiles were written -fPIE wasn't really a thing?  In upstream FreeBSD I think a manpage describing PIE vs PDE vs DSO and the relevant csu files (maybe a 'csu(5)' or 'csu(7)'?) would be beneficial.